packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -104,9 +104,11 @@ export class WorkspaceCommandContribution implements CommandContribution {
                     initialValue: uri.path.base,
                     validate: name => this.validateFileName(name, parent)
                 });
-                dialog.open().then(name =>
-                    this.fileSystem.move(uri.toString(), uri.parent.resolve(name).toString())
-                );
+                dialog.open().then(name => {
+                    renameWidget(this.openerService, uri.parent.resolve(uri.path.base), uri.parent.resolve(name), name).then(() => {
+                        this.fileSystem.move(uri.toString(), uri.parent.resolve(name).toString());
+                    });
```

Comment:
we should:
  - move the file and make sure it worked
 -  update the widget key
 - change the title